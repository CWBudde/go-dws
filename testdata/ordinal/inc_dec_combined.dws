// Test Inc() and Dec() used together
// Task 9.24 and 9.25: Combined ordinal functions

var x: Integer := 10;
PrintLn('Start: ', x);

Inc(x, 5);
PrintLn('After Inc(x, 5): ', x);

Dec(x, 3);
PrintLn('After Dec(x, 3): ', x);

Inc(x);
PrintLn('After Inc(x): ', x);

Dec(x);
PrintLn('After Dec(x): ', x);

// Countdown test
var i: Integer := 5;
var sum: Integer := 0;

while i > 0 do
begin
	sum := sum + i;
	Dec(i);
end;

PrintLn('Countdown sum: ', sum);

// Count up test
i := 1;
var product: Integer := 1;

while i <= 5 do
begin
	product := product * i;
	Inc(i);
end;

PrintLn('Factorial 5: ', product);
