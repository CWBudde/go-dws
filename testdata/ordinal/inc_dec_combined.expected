Start:  10
After Inc(x, 5):  15
After Dec(x, 3):  12
After Inc(x):  13
After Dec(x):  12
Countdown sum:  15
Factorial 5:  120
